### DISCUSSION TITLE
feature request: customizable main menu categories

### DISCUSSION THREAD

**Discussion by antis81** [lxqt/lxqt#2062] [Ideas]
Hey LXQt travellers!

An idea (and feature request) came up yesterday over the discussion how to solve the `Preferences` menu category vs `LXQt Configuration Center` (see [lxqt-config:#759](https://github.com/lxqt/lxqt-config/issues/759)).

Actually what about adding a "Categories" section to the "main menu" configuration dialog? Screenshot should help with the discussion:

![menu_categories](https://user-images.githubusercontent.com/440517/131249345-f8f3820d-d2f1-4b3f-8740-80414a687f27.png)

**Note:** Why would useres want to replace the XDG menu file ("red square")? (IMHO this section could be replaced by the "Categories" section.)

(Found an old issue #255 from 2014 pointing in a similar direction. We should follow - maybe improve - XDG standard here.)

**Comment by stefonarch** [comment]
I'm afraid I do not really understand the "categories feature" you mean, can you be more explicit? The present default categories are translated by lxmenu-data (as I remember).

EDIT: Read the old discussion only now, I understand now. Dragging around items and adding/removing categories would be actually an fail-proof editor for the .menu files. No idea how complicated that would be.


> Why would useres want to replace the XDG menu file ("red square")? (IMHO this section could be replaced by the "Categories" section.)

One of my ideas is this, users could change between some different ones. 
Atm I have the default menu and the one displayed here: https://github.com/lxqt/lxqt-config/issues/759#issuecomment-907648289, and as an idea a 2 section menu with the configuration center and preferences, ommitting the third section LXQt Settings. 

Downside is that there are some other menus present in `/etc/xdg/menus`, `lxde-applications.menu `is  working and shipped by lxmenu-data (there is an orfan attempt to convert that to an lxqt-menu-data), but the others aren't working. 
Better would be a list of menus to select from.

**Reply by tsujan** [reply]
>  would be actually an fail-proof editor for the .menu files

Yes. There's already a feature request for a "menu editor". See https://github.com/lxqt/lxqt/issues/255 and, especially, https://github.com/lxqt/lxqt/issues/255#issuecomment-611804748. A menu editor isn't related to Panel; it's a separate app.

**Reply by tsujan** [reply]
Please also note that categories can be removed by using pcmanfm-qt → Applications but **I don't recommend it at all**. If you want to test it, first back up these directories:

```
~/.local/share/applications
~/.local/share/desktop-directories
~/.config/menus
```

The safest way of adding/removing categories as user is always manual editing but it requires a good knowledge of menu structure.

**Reply by antis81** [reply]
Actually all I am asking for is an option to "hide" the "Preferences" category (to remove redundancy). The only place where the menu can be configured currently is from the panel - IMHO that's okay. A "mockup" should help:

![menu_cat_mockup](https://user-images.githubusercontent.com/440517/131264987-2c6b32a3-a934-4011-84f8-82f3c168a6f6.png)

**Reply by stefonarch** [reply]
> Actually all I am asking for is an option to "hide" the "Preferences" category (to remove redundancy). 

This can already easily be achieved using a custom menu file.
Be aware that atm not everything you see in "Preferences" is also present in "Configuration Center" (actually this should be fixed, by adding an `onlyunallocated` or another section between "LXQt Settings" and "System Settings"  "Other Desktop Settings". Having only one item for the center would be enough then.). 

I'ìm not sure how many user would really use this feature, while a complete drag and drop and rename editor will sure have it's user.
